proxy/upgrade.go

```diff
@@ -61,6 +61,23 @@ type auditLog struct {
 // write data to an auditlog. It will not return until the connection
 // is closed.
 func (p *upgradeProxy) serveHTTP(w http.ResponseWriter, req *http.Request) {
+	// The following 2 checks are based on
+	// https://tools.ietf.org/html/rfc2616#section-14.42
+	// https://tools.ietf.org/html/rfc7230#section-6.7
+	// and https://tools.ietf.org/html/rfc6455 (websocket)
+	if req.Method != http.MethodGet {
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		w.Write([]byte(http.StatusText(http.StatusMethodNotAllowed)))
```

Comment:
i think go server automatically write the status text. Is there a reason to do it here?

Reply:
I just follow the same pattern as it was done before in this function.

Reply:
true, i didn't check the surroundings